### PR TITLE
Simplify spark_eval scripts and improve documentation.

### PR DESCRIPTION
**Pull request by tomwhite**
Should be straightforward, as the changes are very self-contained.

### CODE REVIEW

File: scripts/spark_eval/run_gcs_cluster.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  exit 1
+fi
+
+if [ -z "$API_KEY" ]; then
+  echo "Please set the API_KEY environment variable to a path to the local json file."
+  exit 1
+fi
+
+# Create cluster
+gcloud dataproc clusters create "$GCS_CLUSTER" \
```
Comment (lbergelson) [inline scripts/spark_eval/run_gcs_cluster.sh]: @tomwhite Lets make these auto-deleting clusters now that that's an option.  You can do it by switching to

`gcloud beta dataproc clusters create ...` and adding the new `--max-idle` and `--max-age` parameters.

File: scripts/spark_eval/copy_small_to_hdfs_on_gcs.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+# Copy small data to HDFS on GCS
+
+${GATK_HOME:-../..}/gatk-launch ParallelCopyGCSDirectoryIntoHDFSSpark \
```
Comment (lbergelson) [inline scripts/spark_eval/copy_small_to_hdfs_on_gcs.sh]: Neat, I didn't know about this bashism.   Alternatively you can add gatk-launch to your path.

File: scripts/spark_eval/copy_exome_to_hdfs.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+fi
+
+set -e
+set -x
+
+# Create data directory in HDFS
+hadoop fs -mkdir -p $TARGET_DIR
+
+# Download exome BAM (ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/technical/working/20101201_cg_NA12878/)
+gsutil cp gs://broad-spark-eval-test-data/data/NA12878.ga2.exome.maq.raw.bam - | hadoop fs -put - $TARGET_DIR/NA12878.ga2.exome.maq.raw.bam
```
Comment (lbergelson) [inline scripts/spark_eval/copy_exome_to_hdfs.sh]: Should we use the -m command for parallel download?  That seems to speed things up for me.

Comment (lbergelson) [reply scripts/spark_eval/copy_exome_to_hdfs.sh]: Oh, nm, I didn't read far enough to see that it it was being streamed into hadoop....

File: scripts/spark_eval/README.md [hunk 4]
```diff
@@ -1,69 +1,121 @@
...
 ```bash
-./prep_data_exome_gcs.sh
+./exome_copy_hdfs.sh
 ```
 
 For the whole genome data, run:
 
 ```bash
-./prep_data_genome_gcs.sh
+./genome_copy_hdfs.sh
```
Comment (lbergelson) [inline scripts/spark_eval/README.md]: these names don't seem to match the names of the copy scripts.  I see `copy_genome_to_hdfs.sh`

File: scripts/spark_eval/copy_exome_to_hdfs_on_gcs.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+# Copy exome data to HDFS on GCS
```
Comment (lbergelson) [inline scripts/spark_eval/copy_exome_to_hdfs_on_gcs.sh]: When do you want to use this and when do you want to use `copy_exome_to_hdfs.sh`?

File: scripts/spark_eval/copy_exome_to_hdfs_on_gcs.sh [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+${GATK_HOME:-../..}/gatk-launch ParallelCopyGCSDirectoryIntoHDFSSpark \
```
Comment (lbergelson) [inline scripts/spark_eval/copy_exome_to_hdfs_on_gcs.sh]: Should this be explicitly setting the number of executors?

File: scripts/spark_eval/run_gcs_cluster.sh (hunk 1 above)
Comment (tomwhite) [reply scripts/spark_eval/run_gcs_cluster.sh]: Done

File: scripts/spark_eval/README.md (hunk 4 above)
Comment (tomwhite) [reply scripts/spark_eval/README.md]: Fixed

File: scripts/spark_eval/copy_exome_to_hdfs_on_gcs.sh (hunk 5 above)
Comment (tomwhite) [reply scripts/spark_eval/copy_exome_to_hdfs_on_gcs.sh]: Added some documentation to explain the difference.

File: scripts/spark_eval/copy_exome_to_hdfs_on_gcs.sh (hunk 6 above)
Comment (tomwhite) [reply scripts/spark_eval/copy_exome_to_hdfs_on_gcs.sh]: I don't think it needs to - at least it works fine without.

**Comment by lbergelson** [comment]
@tomwhite Thanks you!

**Pull request merged**